Respect loaded mask cube properly in spectral extraction.

rosteen opened the pull request:
Fixes #3312.

Close https://github.com/spacetelescope/jdaviz/pull/2718

pllim commented on jdaviz/configs/cubeviz/helper.py:
```diff
@@ -22,6 +22,7 @@ class Cubeviz(CubeConfigHelper, LineListMixin):
 
     _loaded_flux_cube = None
     _loaded_uncert_cube = None
+    _loaded_mask_cube = None
```
Maybe give @javerbukh co-author credit given https://github.com/spacetelescope/jdaviz/pull/2718 ?

kecnry reviewed (commented):
The code itself looks good to me - just a few workflow related questions to consider:

Do we want (now or as a a follow-up ticket) to have a dropdown to select the mask cube?  Should the default be to apply the mask or to not apply it?

What happens if the intersection between an aperture and the mask results in no pixels?

Should the mask be used in other plugins as well (aperture photometry, moment map, collapse, etc)?

camipacifici commented:
Visualize the mask cube would be very useful!

gibsongreen commented:
Code seems straightforward, testing this on top of #3307 and the non-finite warning above the Fit Model button goes away with this PR.

In 3307, I was still seeing a model in the spectrum-viewer/table after model fitting, and there wasn't a traceback (but there was a warning was in the log). 

I'll hold off approving for now and re-review if/once the above comments are decided on  and if they go into this PR or follow-ups.

rosteen commented:
@kecnry I just tested, if you define an aperture that is totally covered by the mask, it extracts a flat spectrum at 0 flux.

rosteen commented:
I think I figured out the test failure - the science validation cube has NaNs in the mask/DQ array where the data isn't masked rather than 0s (btw, I did confirm that what we're storing from reading the mask array is what we're showing as the DQ data entry). This is awful, but easy to handle.

rosteen commented:
I removed the lines storing the JWST DQ array as the original mask, and added a note in the docs as well as a Snackbar message indicating the original mask was applied during extraction, to address @camipacifici's concerns.

<img width="991" alt="Screenshot 2024-12-05 at 1 11 40 PM" src="https://github.com/user-attachments/assets/3f5ab411-b04b-4b04-b4c6-5932aa78e4a9">

kecnry reviewed (approved):
Do we have any existing test data that we can send through spectral extraction to add coverage to these logic paths?  Otherwise looks good to me until we can get the full support in to allow mask selection across all relevant plugins.

gibsongreen reviewed (approved):
The recent code updates look good to me and behavior has been maintained since my last review!
